src/services/projects.js

```diff
@@ -318,4 +321,199 @@ module.exports = class ProjectsHelper {
 			throw error
 		}
 	}
+
+	static async submitForReview(resourceId, bodyData, userDetails) {
+		try {
+			let projectDetails = await this.details(resourceId, userDetails.organization_id, userDetails.id)
+
+			if (projectDetails.statusCode !== httpStatusCode.ok) {
+				return responses.failureResponse({
+					message: 'DONT_HAVE_PROJECT_ACCESS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			let projectData = projectDetails.result
+
+			if (projectData.user_id !== userDetails.id) {
+				return responses.failureResponse({
+					message: 'DONT_HAVE_PROJECT_ACCESS',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			let entityTypes = await entityModelMappingQuery.findEntityTypesAndEntities(
+				{
+					model: common.PROJECT,
+					status: common.STATUS_ACTIVE,
+				},
+				userDetails,
+				['id', 'value', 'has_entities', 'validations']
+			)
+
+			// Using forEach instead of for loop for entity type validations
+
+			entityTypes.forEach((entityType) => {
+				const fieldData = projectData[entityType.value]
+
+				if (entityType.validations.required) {
+					let required = utils.checkRequired(entityType, fieldData)
+					if (!required) {
+						throw responses.failureResponse({
+							message: `${entityType.value} not added`,
+							statusCode: httpStatusCode.bad_request,
+							responseCode: 'CLIENT_ERROR',
+							error: utils.errorObject(common.BODY, entityType.value),
+						})
+					}
+				}
+
+				if (entityType.has_entities) {
+					let checkEntities = utils.checkEntities(entityType, fieldData)
+					if (!checkEntities.status) {
+						throw responses.failureResponse({
+							message: checkEntities.message,
+							statusCode: httpStatusCode.bad_request,
+							responseCode: 'CLIENT_ERROR',
+							error: utils.errorObject(common.BODY, entityType.value),
+						})
+					}
+				}
+
+				if (entityType.validations.regex) {
+					let checkRegex = utils.checkRegexPattarn(entityType, fieldData)
+					if (checkRegex) {
+						throw responses.failureResponse({
+							message: `Special characters not allowed in ${entityType.value}`,
+							statusCode: httpStatusCode.bad_request,
+							responseCode: 'CLIENT_ERROR',
+							error: utils.errorObject(common.BODY, entityType.value),
+						})
+					}
+				}
+			})
+
+			if (projectData.tasks.length < 1) {
+				throw responses.failureResponse({
+					message: 'TASK_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+					error: utils.errorObject(common.BODY, common.TASKS),
+				})
+			} else if (projectData.tasks.length < process.env.MAX_PROJECT_TASK_COUNT) {
```

Comment:
@ankit1shahu the condition should be projectData.tasks.length > process.env.MAX_PROJECT_TASK_COUNT